### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by ouziel-slama**
- Fix CORS for OPTIONS requests
- Fix rounding error on normalized quantity
- Use `null` instead of '' for `asset_longname` and `asset_parent` fields
- Correctly catch `ValueError` in unpack endpoint
- Correctly catch `InvalidBase58Error` in compose endpoints
- Correctly catch `BitcoindRPCError` in get transaction info endpoint
- Fix typo in dispenser error messages (`has` -> `have`)
- Fix get balances endpoint when using `sort=asset`
- Catch all errors when using unpack endpoint with invalid data
- Restart RSFetcher when it returns None
- Add `sort` parameter for the get holders endpoint (sortable fields: `quantity`, `holding_type`, and `status`)
- Exclude blocks that are not finished being parsed
- Support the `SENTRY_SAMPLE_RATE` environment variable to set the Sentry sample rate
- Clean up blocks without ledger hash before starting the catch up

------------

* [x] Double-check the spelling and grammar of all strings, code comments, etc.
* [x] Double-check that all code is deterministic that needs to be
* [x] Add tests to cover any new or revised logic
* [ ] Ensure that the test suite passes
* [x] Update the project [release notes](release-notes/)
* [x] Update the project documentation, as appropriate, with a corresponding Pull Request in the [Documentation repository](https://github.com/CounterpartyXCP/Documentation)

### CODE REVIEW

File: counterparty-core/counterpartycore/lib/messages/dispenser.py
Comment (droplister) [inline counterparty-core/counterpartycore/lib/messages/dispenser.py:1]: There are two mentions of "has" asset in this file, see line 232.

File: counterparty-core/counterpartycore/lib/messages/dispenser.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def validate(
         problems.append(f"address doesn't has the asset {asset}")
```
[the remark this answers is not in the document]
Comment (ouziel-slama) [reply counterparty-core/counterpartycore/lib/messages/dispenser.py]: thanks! fixed here https://github.com/CounterpartyXCP/counterparty-core/pull/2700/commits/7bc8f16eb8acd5dfd7331a35587582953e3905e7

**Review by adamkrellenstein — APPROVED** [review]

File: counterparty-core/counterpartycore/lib/api/api_server.py [hunk 2]
```diff
@@ -384,19 +384,35 @@ def init_flask_app():
...
+        methods = ["OPTIONS", "GET"]
         # Add routes
-        app.add_url_rule("/v2/", view_func=handle_route, strict_slashes=False)
         app.add_url_rule(
-            "/v2/blueprint", view_func=handle_doc, methods=["GET"], strict_slashes=False
+            "/v2/",
+            view_func=handle_route,
+            methods=methods,
+            strict_slashes=False,
+            provide_automatic_options=False,
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/lib/api/api_server.py:394]: what does this arg do?

**Review by adamkrellenstein — APPROVED** [review]

File: counterparty-core/counterpartycore/lib/api/api_server.py (hunk 2 above)
Comment (ouziel-slama) [reply counterparty-core/counterpartycore/lib/api/api_server.py:394]: by default Flask manage alone OPTIONS request and doesn't pass them to the application.

**Pull request merged**